custom-recipes/nlp-analysis-ontology-tagging/recipe.json

```diff
@@ -0,0 +1,211 @@
+            "label": "Document dataset",
+            "description": "Dataset with a column containing the text documents to tag",
+            "arity": "UNARY",
+            "required": true,
+            "acceptsDataset": true
+        },
+        {
+            "name": "Ontology dataset",
+            "label": "Ontology dataset",
+            "description": "Dataset with columns for tags, keywords and categories",
+            "arity": "UNARY",
+            "required": true,
+            "acceptsDataset": true
+        }
+    
+    ],
+
+    "outputRoles": [
+        {
+            "name": "Tagged documents",
+            "label": "Tagged documents",
+            "description": "Dataset with the input documents and their assigned tags",
+            "arity": "UNARY",
+            "required": true,
+            "acceptsDataset": true
+        }
+  
+    ],
+    
+    "selectableFromDataset": "Document dataset",
+    "selectableFromDataset":"Ontology dataset",
+
+    "paramsPythonSetup": "pick_languages.py",
+    "params": [
+        
+         {
+            "name": "separator_document_dataset",
+            "label": "Document dataset",
+            "type": "SEPARATOR"
+        },
+        
+        
+        {
+            "name": "text_column",
+            "label": "Text column",
+            "type": "COLUMN",
+            "description": "Column containing the text documents",
+            "mandatory": true,
+            "columnRole": "Document dataset",
+            "allowedColumnTypes": [
+                "string"
+            ]
+        },
+
+        {
+            "name": "language",
+            "label": "Language",
+            "type": "SELECT",
+            "getChoicesFromPython": true,
+            "defaultValue": "en"
+        },
+        {
+            "name": "language_column",
+            "label": "Language column",
+            "type": "COLUMN",
+            "description": "Column containing ISO 639-1 language codes",
+            "columnRole": "Document dataset",
+            "allowedColumnTypes": [
+                "string"
+            ],
+            "visibilityCondition": "model.language == 'language_column'"
+        },
+        
+        {
+            "name": "separator_ontology_dataset",
+            "label": "Ontology dataset",
+            "type": "SEPARATOR"
+        },
+        
+        {
+            "name": "tag_column",
+            "label": "Tag column",
+            "type":"COLUMN",
+            "mandatory":true,
+            "columnRole":"Ontology dataset",
+            "allowedColumnTypes": [
+                "string"
+            ]
+        },
+        
+       
+         
+        {
+            "name":"keyword_column",
+            "label":"Keyword column",
+            "type":"COLUMN",
+            "mandatory": true,
+            "columnRole":"Ontology dataset",
+            "allowedColumnTypes": [
+                "string"
+            ]
+            
+        },
+        
```

Comment:
Sorry to nitpick here - Can you lint the JSON so that no spaces are left? I can show you tomorrow how to do it automatically in VScode

Reply:
done in f2676d1